pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -370,6 +379,55 @@ func (b *EndpointBuilder) BuildClusterLoadAssignment(endpointIndex *model.Endpoi
 	return l
 }
 
+func (b *EndpointBuilder) generateToServiceWaypoint() []*LocalityEndpoints {
+	if b.proxy.IsWaypointProxy() || b.dir != model.TrafficDirectionOutbound || !features.EnableHBONE {
+		return nil
+	}
+	waypoint := b.toServiceWaypoint()
+	if waypoint == "" {
+		return nil
+	}
+
+	serviceVIPs := b.service.ClusterVIPs.GetAddressesFor(b.clusterID)
```

Comment:
When would this happen ? Headless services ? Comment would be good.

Reply:
Will add a comment. It shouldn't ever happen if we make it to this point in the code. We don't build EDS for headless, it's ORIGINAL_DST.

Reply:
I think we have some code to deal with headless ( listeners or filters ) - my point is really that we should have comments in the code and docs explaining this clearly. It is not obvious what happens or what we want to happen.